src/main/java/org/generationcp/middleware/api/ontology/OntologyVariableServiceImpl.java

```diff
@@ -27,4 +40,101 @@ public OntologyVariableServiceImpl(final HibernateSessionProvider sessionProvide
+		// Get the existing trait variables
+		final Map<Integer, Variable> variablesMap = this.daoFactory.getCvTermDao().getVariablesWithFilterById(variableFilter);
+		// Create ontology methods for analysis names if not yet present, will also return methods if already present
+		final Map<String, CVTerm> methodsMap = this.createOntologyMethodsIfNecessary(analysisNames);
+		// Check if the analysis variables associated to trait variables are already present
+		final MultiKeyMap existingAnalysisMethodsOfTraitsMap = this.daoFactory.getCvTermRelationshipDao()
+			.retrieveAnalysisMethodsOfTraits(variableIds, methodsMap.values().stream().map(CVTerm::getCvTermId).collect(
+				Collectors.toList()));
+
+		// Create analysis variable for every trait and analysis methods combination
+		for (final Map.Entry<Integer, Variable> variableEntry : variablesMap.entrySet()) {
+			for (final String analysisName : analysisNames) {
+				final CVTerm method = methodsMap.get(analysisName);
+				// If analysis variable already exists for specific trait, do not create new, just return the existing id of analysis variable
+				if (existingAnalysisMethodsOfTraitsMap.containsKey(variableEntry.getKey(), method.getCvTermId())) {
+					createdAnalysisVariables.add(
+						(Integer) existingAnalysisMethodsOfTraitsMap.get(variableEntry.getKey(), method.getCvTermId()));
+				} else { // else, create new analysis variable
+					createdAnalysisVariables.add(
+						this.createAnalysisStandardVariable(variableEntry.getValue(), method, variableType));
+				}
+			}
+		}
+		return createdAnalysisVariables;
+	}
+
+	@Override
+	public Multimap<Integer, VariableType> getVariableTypesOfVariables(final List<Integer> variableIds) {
+		final Multimap<Integer, VariableType> variableTypesMultimap = ArrayListMultimap.create();
+		final List<CVTermProperty> properties =
+			this.daoFactory.getCvTermPropertyDao().getByCvTermIdsAndType(variableIds, TermId.VARIABLE_TYPE.getId());
+		for (final CVTermProperty property : properties) {
+			variableTypesMultimap.put(property.getCvTermId(), VariableType.getByName(property.getValue()));
+		}
+		return variableTypesMultimap;
+	}
+
+	private Integer createAnalysisStandardVariable(final Variable traitVariable, final CVTerm method, final String variableType) {
+		String analysisVariableName = traitVariable.getName() + "_" + method.getName();
+
+		/** It's possible that the analysisVariableName to be saved is already used by other variables,
+		 * in that case append _1 to the name to make sure it's unique. This is the same logic found in
+		 * {@link org.generationcp.commons.service.impl.BreedingViewImportServiceImpl#createAnalysisVariable(org.generationcp.middleware.domain.dms.DMSVariableType, java.lang.String, org.generationcp.middleware.domain.oms.Term, java.lang.String, int, boolean)}
+		 **/
+		final CVTerm existingTerm = this.daoFactory.getCvTermDao().getByNameAndCvId(analysisVariableName, CvId.VARIABLES.getId());
```

Comment:
Would it be better to use `CvTermDao#getTermsByNameOrSynonym` to check if variable name exists?